slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java

```diff
@@ -65,6 +60,30 @@ public boolean checkGenesisValidatorsRootAndInsertIfEmpty(final Bytes32 genesisV
     }
   }
 
+  private Bytes32 insertAndCacheGVR(final Bytes32 genesisValidatorsRoot) {
+    var cachedGVR = cachedGVRRef.get();
+    if (cachedGVR == null) {
+      LOG.debug("No cached GVR found, fetching from database");
+      synchronized (lockForCachedGVR) {
```

Comment:
Can one of the AtomicReference atomic operations be used to update the value to avoid the synchronized block?